CustomBatteries/API/CBModelData.cs

```diff
@@ -0,0 +1,44 @@
+﻿
+using UnityEngine;
+
+namespace CustomBatteries.API
+{
+    public abstract class CBModelData
```

Comment:
Instead of making this an `abstract` class with `virtual` properties, I suggest we make it normal class with `{ get; set; }` on all the properties.

That should take care of situations where the getter override has to execute heavy code more than once just to check if it has a value.

Reply:
got no problem with that if you want to do it.